shared/src/main/java/uk/gov/di/authentication/shared/validation/MfaCodeValidator.java

```diff
@@ -40,5 +40,5 @@ void resetCodeIncorrectEntryCount(MFAMethodType mfaMethodType) {
         codeStorageService.deleteIncorrectMfaCodeAttemptsCount(emailAddress, mfaMethodType);
     }
 
-    public abstract Optional<ErrorResponse> validateCode(String code);
+    public abstract Optional<ErrorResponse> validateCode(String code, String profileInformation);
```

Comment:
Two little questions here:

1) I wonder if there is an alternative name that would be more descriptive of the purpose? I guess I think of this as something like `optionalSecretOverride`. The drawback with calling it `profileInformation` is that in the although the context is clear where you're using it, in other lambda (verify code) if we start using the same abstract class, that variable name will be less meaningful

2) I also wonder if there is any reason not to just overload the method so that for validators where this field will never make sense, we can just ignore it and continue to call `validateCode(String code)` i.e. keep a second method that just does something like this:

```
    Optional<ErrorResponse> validateCode(String code){
        return validateCode(code, null);
    }
```

It's not so much that this makes any functional difference; but maybe if we then only called validate with the extra parameter for auth app registration scenario, it could make the handler logic that calls `validateCode` more explicit. But that's probably just personal preference/way I'd read the code, so may not seem better for other people.

Reply:
Even I don't think 2 makes sense after a little reflection....!

Reply:
Agreed. I've tried to make it clearer by creating an abstract class for the Code Request. Let me know what you think